### DISCUSSION TITLE
Integration of large dataset fails due to Sparse Matrix Error

### DISCUSSION THREAD

**Discussion by dtm2117** [satijalab/seurat#5221] [Q&A]
Hello,

I am trying to perform integration of a large dataset (61 samples), and it fails at the IntegrateData step

`RNA.combined <- IntegrateData(anchorset = RNA.anchors, new.assay.name = "RNA_INTEGRATED", dims = 1:50, normalization.method = "LogNormalize")
`

I've used the RPCA workflow, but would like to use the log normalization so I can annotate with the BM reference provided. 

Here is the error i receive after many hours of running:
`Error in subCsp_rows(x, i, drop = drop) : 
  Cholmod error 'problem too large' at file ../Core/cholmod_sparse.c, line 89 `

I have access to 500GB of ram, so I'm wondering why i can't expand this matrix. Any help would be extremely appreciated. 

Best,
David

**Comment by timoast** [comment]
You could try running `IntegrateEmbeddings()` rather than `IntegrateData()`, which requires a much smaller matrix

**Reply by dtm2117** [reply]
Thanks for your reply! I'm surprised that the error pops up, because I believe the cell count isn't so high that the matrix should cause this error. (~350K cells) 

Can I utilize the same downstream processing post IntegrateEmbeddings() ?

**Reply by timoast** [reply]
`IntegrateEmbeddings()` will correct the PCA coordinates, you can use the integrated PCA for downstream steps like creating a UMAP visualization and cell clustering.
